### ISSUE TITLE
WebRTC not working

### ISSUE THREAD

**Issue by jcardila** [RocketChat/Rocket.Chat#2122]
I can't get webrtc calls to work, tried on own server and on RC demo server and couldn't get them to work. Both pass the webrtc test except for IPv6 and a warning in Reflexive connectivity. Tried connected in the same network and on different networks.

**Comment by Sing-Li** [comment]
I have experienced the same thing with the demo server, only since the CPU prob.  @rodrigok   Do you think  it be related to this change?  tia

https://github.com/RocketChat/Rocket.Chat/commit/cd829c3f92ab6ef2bca8270afbfddadf2ad7eacc

**Comment by rodrigok** [comment]
@Sing-Li probably yes, we need some time to improve that

**Comment by bott0r** [comment]
Indeed, since i updated to 0.17.0 WebRTC does not seem to work.

I am aware that the webRTC functions are not really a priority at the moment. But if there would be a possibility to include a settings button to change voice activation levels, or include a utilize push to talk button, that would be greatly appreciated.

**Comment by realAkhmed** [comment]
Same here. WebRTC problems in 0.17.0
@bott0r what was the previous version that you used when WebRTC worked?

**Comment by bott0r** [comment]
@realAkhmed for me it was 0.15 or something. I am currently on the developer version.

**Comment by xenithorb** [comment]
Im having the same issues... Broken since Feb? I take it no one actually uses this to talk to people then

**Comment by thomas-mangin** [comment]
Yes, it is broken (which is annoying). It does not seems to be tested when releases are done.

**Comment by xenithorb** [comment]
Are there plans to get this working again? Or drop the feature advertising from the home page? 

I large part of why I spent the time to deploy this was to do WebRTC calling...

**Comment by jeromeza** [comment]
Same here - been playing around trying to get this working for a day or two now - only to find out its still broken. What version is the demo running as it works on there?

**Comment by xenithorb** [comment]
### Perhaps this will help some people: 

xenithorb 2:39 PM 
@frdmn: I'm sorry, but I just realized that WebRTC audio does work between two of the reference electron clients. It seems my earlier failures to get it to work were due to doing LAN->LAN testing with a server that existed on the public internet. We could not however get video to work. Electron also needs to be able to anchor to an audio device, like firefox/chrome. Otherwise it worked really well 🙂
Perhaps the information warning should include something about not trying to contact in-LAN other clients while using an external server? Perhaps the default TURN/STUN settings mess the ability to do that up?

----
I wrote that in #support on the demo server earlier.

**Comment by jeromeza** [comment]
I still can't get audio  / video to work.

My server is external from my network. Clients are also on different networks.

I just get call popups, answer and nothing happens.

**Comment by thomas-mangin** [comment]
Same here - all clients and the server are on real word IP (ie: no NAT anywhere). The server is however firewalled only allowing port 443 (the TURN/STUN server is on another IP).

**Comment by thomas-mangin** [comment]
0.32 works for me.

**Comment by jmls** [comment]
we can't seem to get this working at all - we're running 0.33 in docker . If I call a friend, nothing happens. if he calls me, the popup appears  - but I click accept nothing happens

![image](https://cloud.githubusercontent.com/assets/929523/15964363/14d97200-2f10-11e6-8573-138046053109.png)

**Comment by thomas-mangin** [comment]
It has all the characteristics of a NAT issue. Make sure STUN is correctly setup.

**Comment by jmls** [comment]
ah - ok - do we need to set up our own STUN server then ?

**Comment by xenithorb** [comment]
Thing is, I've never been able to get "Reflexive connectivity" to work tested by http://test.webrtc.org

That seems to be causing some issues, other issues like when a call doesn't work, the other party isn't really notified if their client thinks they're in a call are clearly event-driven bugs of the product. There are a lot of errors here, and it's only compounded by trying to be compatible with many browsers (Which should be commended, I'm just saying though) 

On Linux, with the reference client, I am able to make calls (and do frequently), but only works on occasion from recent versions of chromium, and doesn't seem to work at all on Firefox.

Screen sharing is a really odd one, too. In this case it doesn't work in the reference client because they haven't yet figured out how to get the extension to work in Electron. Firefox doesn't show a video feed because the selector never appears. And Chromium you can select the viewport to share, and that work, to include an image of your desktop where you would expect video to be.... But the other end only sees black. 

Video is the same way, I imagine for the same reasons. Black on the other end.

**Comment by thomas-mangin** [comment]
@jmls
```
root@stun:~# dpkg -l | grep -i turn
ii  resiprocate-turn-server             1:1.9.7-4~ubuntu14.04.1          amd64        reSIProcate SIP stack - ICE/TURN server
```

**Comment by jmls** [comment]
@thomas-mangin thanks for the info - However, I would have thought that if this were needed it would have been supplied and installed on the official rocketchat docker image though .. or am I being thick ?

**Comment by xenithorb** [comment]
I'm personally targeting `coturn` through an ansible role I'm in the process of writing

**Comment by thomas-mangin** [comment]
@jmls I installed rocket-chat 'by hand' (well, with my own scripts) so I have no idea what the docker image looks like but I would not assume they done it.

**Comment by Sing-Li** [comment]
### New chat-centric non-intrusive solution testing this week.

On https://demo.rocket.chat

When you are in a Direct Message or Private Group. You will notice a new video button on the toolbar , just click it anytime to start a video conference with all those in the room

**Comment by jmls** [comment]
@Sing-Li : is this something that's going to replace the existing audio/video ? (ie should we be wasting time trying to solve the problems that we are having now on 0.33?)

**Comment by Sing-Li** [comment]
Very likely.  But we're not ready to make any official statements just yet.

Meanwhile, please try things out and give us as much feedback as possible.  We expect the new solution's stability and robustness to be improve several folds, and hopefully the learnt-from-feedback UI/Ux is actually superior as well.

**Comment by jmls** [comment]
excellent news. We will want it to work in docker behind an nginx proxy - so that's something to bear in mind :)

It would also be very useful to have an icon to start the conference with audio-only - I hate having to scrabble to turn the video off ...

**Comment by Sing-Li** [comment]
Yes on both counts - not immediately on switch-over ... but already on TODO  :)

**Comment by jmls** [comment]
cool. how long do we have to dream before this becomes a reality - unofficially of course ...

**Comment by Sing-Li** [comment]
haha ... you might have to DM one of us on the demo to get an OTR message on a vapor estimate 😀

**Comment by jmls** [comment]
ohhh! testing otr as well ;) will do ..

**Comment by MarPiRK** [comment]
I had similar problems with STUN/TURN and found list of pretty reliable servers somewhere on Stack Overflow, which work for me. Here it is:

```stun:stun01.sipphone.com,stun:stun.ekiga.net,stun:stun.fwdnet.net,stun:stun.ideasip.com,stun:stun.iptel.org```

**Comment by jmls** [comment]
you tease !!

**Comment by MarPiRK** [comment]
Sorry, sth went wrong with Markdown

**Comment by jmls** [comment]
any news on the new feature ?

**Comment by xenithorb** [comment]
It was released (in beta) as a feature on the stable 0.34.0 version. Check the changelog.

**Comment by propertunist** [comment]
i'm testing the current public version of rocket.chat here and also unable to get any webrtc connections going. i am seeing the same behavior as above - the call initiates, but no streams are shared, so no call occurs. how do i find relevant error messages from rocketchat to help diagnose the issue? i see nothing in the terminal i am running rocket chat from. thanks

**Comment by johnfelipe** [comment]
![webrtc](https://cloud.githubusercontent.com/assets/428820/16509749/a862cab2-3f05-11e6-80b5-2a029d45ff63.png)

Versión 51.0.2704.106 m (64-bit) of chrome

nothing happend

![firefox](https://cloud.githubusercontent.com/assets/428820/16509760/ba8c3908-3f05-11e6-9c96-ac7da5f0eadd.png)

![test](https://cloud.githubusercontent.com/assets/428820/16509854/ac9068a0-3f06-11e6-98a2-370f01aae9ab.png)

**Comment by propertunist** [comment]
@asthomasdk - any insights you can provide as to how you got even the audio to work would be helpful to me as i have never seen the audio or video to work in my rocketchat install. thanks

**Comment by hadifarnoud** [comment]
I can't figure out why this isn't working for me. tried WebRTC settings people said here too
<img width="951" alt="screen shot 1395-12-21 at 8 51 04 pm" src="https://cloud.githubusercontent.com/assets/589808/23825319/ac6608fe-069c-11e7-9284-be78db92726d.png">

**Comment by hadifarnoud** [comment]
this issue been open for months!!!

**Comment by frdmn** [comment]
Because we don't prioritize issues based only by the creation date. Read up the relevant issues regarding WebRTC and you'll understand why this one is still open.

Von meinem iPhone gesendet

**Comment by hadifarnoud** [comment]
I've read #4846 but still don't know why this issue is still open. can you please tell us why @frdmn ?

**Comment by frdmn** [comment]
@hadifarnoud Simply because no-one came up with a proper and advanced solution for this. If you can help you, you're very welcome to do so.

**Comment by hadifarnoud** [comment]
I get this error when clicking on mic.
```
getUserMedia() no longer works on insecure origins. To use this feature, you should consider switching your application to a secure origin, such as HTTPS. See https://goo.gl/rStTGz for more details.
```

I guess that's because I have RC on http (without SSL)

**Comment by alirezaimi** [comment]
I installed  0.58.2 and mic and video call and direct call not work in android or desktop or web interfaces .

**Comment by Alseenrodelap** [comment]
The call and video call functions still don't work. Do I have to setup something, or open ports?

**Comment by HammyHavoc** [comment]
Yep, having issues with this too. I click accept on the call and then nothing.

**Comment by bse5** [comment]
same here ... Accept and nothin happens

**Comment by TerrenceLP** [comment]
Unable to get property 'user' of undefined or null reference
SCRIPT438: Object doesn't support property or method 'createMediaStreamDestination'

This using a Paid TURN/stun twilio servers

**Comment by lostdesign** [comment]
Where would that button be after all? I cannot find it in the ui?

**Comment by hugocostadev** [comment]
Hi there, thanks for the contribution! 🚀 💯 

Sorry, this error does not happen on the latest and maintained releases.

[Rocket.Chat](https://rocket.chat/)  provides support for, at least, 3 months for each release and 6 months to the latest releases of each major version

I would suggest you update your Rocket.Chat version to the latest stable version, which you can find in [Releases Page](https://github.com/RocketChat/Rocket.Chat/releases) and, check it [HERE](https://docs.rocket.chat/resources/get-support/enterprise-support#rocket.chat-versions) the table with versions and End of Life dates. 

---

**Questions? Help needed? Feature Requests?** 

- Join our [Open Server](https://open.rocket.chat/) in the #support channel and feel free to raise a question
- Join our [Community Forum](https://forums.rocket.chat/) and search/create a post there
